### ISSUE TITLE
Any other services worth implementing?

### ISSUE THREAD

**Issue by rvalitov** [McCzarny/upload-image#19]
I'm curious what other services and alternatives for Imgbb can be implemented?
I found this one https://www.imghippo.com/api-info but it's not better than Imgbb.
Perhaps there are some others who offer more free space?
This post is for sharing the links and ideas.

**Comment by McCzarny** [comment]
I hoped that some day someone will create a request for another service. When I needed file uploading functionality imgbb was ranked at the top so I picked it. Imgur is something that I see quite often, but after reading about it, it seems not fully realible:
```
It’s best known among individuals using it for temporary image hosting for social shares.
Though some Redditors confirm that their files remain on the platform for years,
Imgur clearly offers no guarantees, especially if a user breaches their terms of service (including using Imgur commercially).
```
I would be happy to add another upload method, but I have no clue which one would be needed.

**Comment by rvalitov** [comment]
Indeed, Imgur is not reliable. Their website opens from my browser, but from hosting nothing works:
```json
{"data":{"error":"Imgur is temporarily over capacity. Please try again later."},"success":false,"status":403}
```
It seems they block VPN and (some?) hosting providers...

**Comment by rvalitov** [comment]
Cloudinary seems to be very interesting. The free plan has credits that can store minimum 2Gb of images. That seems to be more than enough as a temporary space for screenshots. What do you think about it?

**Comment by rvalitov** [comment]
Another idea to upload to FTP as to very affordable space of any hosting provider.

**Comment by McCzarny** [comment]
Hello @rvalitov , sorry for the late response. At first FTP was tempting idea, but on the other hand it would be easier to run ftp with a shell command manually than setting up an extension. Do you think that Cloudinary is currently better than imgbb? There is so much stuff I want to do with so little time, that I need to prioritize tasks. I'm happy to implement it, but if there is no explicit request to make it work, then I would rather put it on the todo list.

**Comment by rvalitov** [comment]
Thank you for your comment about my ideas! I'm happy that you're open for the feedback and work on this project and improve it!
It's a really useful action! And as I use it more and more for my repositories I face a problem with lack of free space available at imgbb. The image expiration and deletion features that I was asking for are great but even they do not help me anymore. Imgbb provides only 50 Mb for free, and it seems this limit is too small for me now. Simultaneous PRs in my multiple repositories eat up all this space. I don't want to buy a paid plan at imgbb as I use it only for my PRs as a temporary screenshots hosting. So Cloudinary with its 2Gb+ free space looks very attractive for me. I think it will be more than enough to cover all my current and future  needs. I suppose implementing Cloudinary support will give a solid boost to this project and make it more universal and available for greater number of repositories and users.

P.S. If you have a TODO list which you are comfortable to make a publicly available - you're welcome to share it as feature issues or in the projects section here or in milestones or in any other section that you think is more suitable for that. I already saw 2 issues here. I could help with some minor things by making PRs, so you can focus on Cloudinary. If there are some features that you think must be implemented before adding Cloudinary support - please point that out. I really love your action and would like to contribute. I completely understand your lack of time, I'm also very busy right now and can't promise that I will be able to contribute as quickly as and as much as I would like, but at least seeing some roadmap or milestones will be great, so that I could pick up some issue to implement when I will get a minute of free time!

**Comment by McCzarny** [comment]
Based on what you wrote, the difference in available space is a good reason to implement the new service. I will try to at least start it during the weekend to check what is possible with Cloudinary API.

Concerning the TODO list - I have a GitHub project with a kanban containing tasks from different projects where I want to add something or create new things. On top of that there are some unplanned things for sudden ideas ;)

**Comment by rvalitov** [comment]
Thank you so much! 🥰 
I hope the Cloudinary API will not cause any troubles.

**Comment by McCzarny** [comment]
@rvalitov added initial support for Cloudinary. It's possible to upload images. I think it should be possible to implement delete action for Cloudinary, similar as it is for imgbb. We can leave this issue open in case there will be another API worth implementing.

**Comment by rvalitov** [comment]
Thank you so much for the awesome news and great work! I will test it next week!
